include/swift/AST/DiagnosticsSema.def

```diff
@@ -3785,9 +3785,9 @@ WARNING(guard_always_succeeds,none,
 ERROR(expression_unused_closure,none,
       "closure expression is unused", ())
 ERROR(expression_unused_function,none,
-      "expression resolves to an unused function", ())
+      "function is unused", ())
 WARNING(expression_unused_lvalue,none,
-        "expression resolves to an unused %select{variable|property|subscript}0", (unsigned))
+      "result of %select{variable|property|subscript}0 is unused", (unsigned))
```

Comment:
A couple of more options 
* `expression results in an unused %select{variable|property|subscript}0`
* `%select{variable|property|subscript}0 produced from expression is unused`

Reply:
[Thinking out loud]

For the variable case, it reads oddly IMO. E.g. "result of variable is unused" seems weird (what does it mean for a variable to have a result?), it would be simpler to have "variable is unused".

For the subscript and property cases, the "result" corresponds to some evaluation, i.e. we are accessing the subscript/property (thanks @brentdax for suggesting the word "access"). It's not a result of the property itself, it's a result of accessing the property (or, more pedantically, the result of evaluating the getter of the property). So how about the following options:

1. "{property|subscript} is accessed but the result is unused"
2. "result of {property|subscript} access is unused"

I'm leaning more towards 2 (because it's shorter but has all the key terms). Thoughts?

Reply:
Personally, I slightly prefer the first of your two alternatives. "But" emphasizes the contrast between the clauses it relates, so the first one more clearly suggests that the compiler thinks there's something contradictory about accessing a variable but not using its result. This is a pretty minor preference, though—both of them are fine.

Reply:
I agree with @brentdax about prefering the first alternative, although it a bit longer just seems to read better :)

Reply:
Ok, sure, we can go with the first one then. 😄

Reply:
Thanks for the input! :) I changed it to 
- `function is unused`
- `variable is unused`
- `property/subscript is accessed but result is unused`

Reply:
I wonder if it reads better without `but` - `result of property/subscript access is unused`, WDYT?

Reply:
Yeah, I was leaning towards that, but Luciano and Brent seem to like the first one better. Since we are at 2 people in favor of either option, I think it's okay if we go with option 1 (which Minhyuk already implemented). 🙂

Reply:
Ah sorry, I missed it. Let’s use the version with “but” then